src/extension/intellisenseHelper.ts

```diff
@@ -16,25 +16,25 @@ export class IntellisenseHelper {
     /**
      * Helper method that configures the workspace for Salsa intellisense.
      */
-    public static setupReactNativeIntellisense(): void {
+    public static setupReactNativeIntellisense(): Q.Promise<void> {
         // Telemetry - Send Salsa Environment setup information
-        let tsSalsaEnvSetup = TelemetryHelper.createTelemetryEvent("RNIntellisense");
+        const tsSalsaEnvSetup = TelemetryHelper.createTelemetryEvent("RNIntellisense");
         TelemetryHelper.addTelemetryEventProperty(tsSalsaEnvSetup, "TsSalsaEnvSetup", !!process.env.VSCODE_TSJS, false);
         Telemetry.send(tsSalsaEnvSetup);
 
-        Q({})
+        const configureWorkspace = Q({})
             .then(() => TsConfigHelper.allowJs(true))
             .then(() => TsConfigHelper.addExcludePaths(["node_modules"]))
-            .then(() => IntellisenseHelper.installReactNativeTypings())
-            .done();
+            .then(() => IntellisenseHelper.installReactNativeTypings());
 
         // The actions taken in the promise chain below may result in requring a restart.
-        Q(false)
+        const configureTypescript = Q(false)
             .then((isRestartRequired: boolean) => IntellisenseHelper.enableSalsa(isRestartRequired))
```

Comment:
I don't want to add more noise to this PR, so I'm just adding a TODO:
        /* TODO: Refactor this code to
            Q.all([enableSalsa(), installTypescript(), configureWorkspace()])
            .then((result) => warnIfRestartIsRequired(result.any((x) => x)))
        */